presto-main/src/main/java/com/facebook/presto/util/LongValueCompressor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class LongValueCompressor
```

Comment:
I can in-line this. It was even my first approach, but I had a bug there. Writing tests for an operator in isolation is not an easy task to do. So this is why I decided to extract this. 

It is easy to do a mistake when doing bitwise operation so I wanted to be sure it is well tested, as such bugs could really difficult to fix when spotted at production (it could end up with incorrect result set).

Regarding performance, IMO it is premature-optimization, such simple abstraction should be easily optimized by HotSpot.

Your call, should I in-line this?